### PR TITLE
salmid 0.1.23

### PR DESCRIPTION
**Pull request by dmohadjel**
- [x] Have you followed the [guidelines for contributing](https://github.com/brewsci/homebrew-bio/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/brewsci/homebrew-bio/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source FORMULA`, where `FORMULA` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict FORMULA` (after doing `brew install FORMULA`)?

-----

### CODE REVIEW

**Comment by tseemann** [comment]
@dmohadjel this has failed because it looks like SalmID has changed its whole directory structure and it looks to be pip compatible now. 
https://github.com/hcdenbakker/SalmID

**Comment by dmohadjel** [comment]
@tseemann What changes should I make to the file to accommodate this?

**Comment by tseemann** [comment]
I think it becomes more complicated now. 
need to use the `virtualenv` support for python that brew has.
see https://github.com/Homebrew/brew/blob/master/docs/Python-for-Formula-Authors.md

**Comment by sjackman** [comment]
Try…
```ruby
system "pip3", "install", "--prefix=#{libexec}", "."
```
See `transabyss` and `tigmint`.

**Comment by dmohadjel** [comment]
The error in CircleCI:
```
==> /home/linuxbrew/.linuxbrew/Cellar/salmid/0.1.23/bin/SalmID.py --version
Error: brewsci/bio/salmid: failed
An exception occurred within a child process:
  Errno::ENOENT: No such file or directory - /home/linuxbrew/.linuxbrew/Cellar/salmid/0.1.23/bin/SalmID.py
```

@sjackman I tried making the change you suggested using `pip3` but am arriving to the same error when I run `brew test salmid`. There is no `SalmID.py` file in `/home/dmohadjel/.linuxbrew/Cellar/salmid/0.1.23/bin/SalmID.py`, but it can be found in `/home/dmohadjel/.linuxbrew/Cellar/salmid/0.1.23/libexec/bin/SalmID.py`. When I run...
```
/home/dmohadjel/.linuxbrew/Cellar/salmid/0.1.23/libexec/bin/SalmID.py --version
```
the result is: 
```
Traceback (most recent call last):
  File "/home/dmohadjel/.linuxbrew/Cellar/salmid/0.1.23/libexec/bin/SalmID.py", line 6, in <module>
    from salmid.core import main
ModuleNotFoundError: No module named 'salmid'
```

**Comment by sjackman** [comment]
You need to create a wrapper shell script using either `bin.env_script_all_files` or `bin.write_env_script` in `bin` that sets the `PYTHONPATH` environment variable, so that the Python script can find its other Python files. See `tigmint` for example and others.
https://github.com/brewsci/homebrew-bio/blob/656215c3d8518007ebc046428f92332bb6b08f40/Formula/tigmint.rb#L37

**Comment by jonchang** [comment]
```
Error: 1 problem in 1 formula detected
brewsci/bio/salmid:
  * stable version should not decrease (from 0.122 to 0.1.23)
```

You may need to bump the `version_scheme`.

File: Formula/salmid.rb [hunk 1]
```diff
@@ -14,7 +14,13 @@ class Salmid < Formula
   depends_on "python"
 
   def install
+    ENV.prepend_path "PATH", libexec/"bin"
+    xy = Language::Python.major_minor_version "python2"
```
Comment (sjackman) [inline Formula/salmid.rb]: This should probably be `python3`.

File: Formula/salmid.rb [hunk 2]
```diff
@@ -14,7 +14,13 @@ class Salmid < Formula
...
+    ENV.prepend_path "PYTHONPATH", libexec/"lib/python#{xy}/site-packages"
+    system "pip3", "install", "--prefix=#{libexec}", "."
+    inreplace Dir["SalmID*py"], "#!/usr/bin/env python", "#!#{Formula["python3"].bin}/python3"
     prefix.install Dir["*"]
+    prefix.env_script_all_files libexec/"bin", :PYTHONPATH => Dir[libexec/"lib/python*/site-packages"].first
```
Comment (sjackman) [inline Formula/salmid.rb]: Should `prefix` be `bin`?

**Comment by dmohadjel** [comment]
Still having this issue on CircleCI
```
==> /home/linuxbrew/.linuxbrew/Cellar/salmid/0.1.23/bin/SalmID.py --version
Error: brewsci/bio/salmid: failed
An exception occurred within a child process:
  Errno::ENOENT: No such file or directory - /home/linuxbrew/.linuxbrew/Cellar/salmid/0.1.23/bin/SalmID.py
```

**Comment by sjackman** [comment]
Are you able to reproduce this issue locally?
Do both the file `/home/linuxbrew/.linuxbrew/Cellar/salmid/0.1.23/bin/SalmID.py` and its interpreter exist? (the file specified by the shebang line `#!`)

**Comment by dmohadjel** [comment]
> Do both the file `/home/linuxbrew/.linuxbrew/Cellar/salmid/0.1.23/bin/SalmID.py` and its interpreter exist? (the file specified by the shebang line `#!`)

The interpreter exists, but the SalmID.py does not exist in `/home/linuxbrew/.linuxbrew/Cellar/salmid/0.1.23/bin/`.

**Comment by sjackman** [comment]
> SalmID.py does not exist in /home/linuxbrew/.linuxbrew/Cellar/salmid/0.1.23/bin/

Can you troubleshoot that issue and hopefully fix it?

**Comment by sjackman** [comment]
Merged. Thanks for fixing this issue, Dara! 🥇

**Pull request closed without merging**